v7/_posts/design/2018-09-26-$expandForV4.01.md

```diff
@@ -0,0 +1,112 @@
+---
+layout: post
+title: "6.2 Design doc for $expand in context-url for OData V4.01"
+description: "Design doc for $expand in context-url for OData V4.01"
+category: "6. Design"
+---
+
+# $expand in Context Url
+
+## Introduction
+In form of metadata, context url in response from OData-compliant service provides a way to describe the response payload, and is used as control information to facilitate response processing by the client. 
+
+Here is an example of a simple query request and resulting response containing context url:
+
+Request url: http://services.odata.org/V4/TripPinService/People('russellwhyte')?$select=FirstName,LastName
+
+Response:
+```
+{
+    "@odata.context": "http://services.odata.org/V4/TripPinService/$metadata#People(FirstName,LastName)/$entity",
+    "@odata.id": "http://services.odata.org/V4/TripPinService/People('russellwhyte')",
+    "@odata.etag": "W/\"08D62407F53DE9ED\"",
+    "@odata.editLink": "http://services.odata.org/V4/TripPinService/People('russellwhyte')",
+    "FirstName": "Russell",
+    "LastName": "Whyte"
+}
+```
+As shown above, @odata.context annotation specifies the context url, and provides the following machine-readable description for response data:
+- It is regarding an entity from the "People" entity set,
+- It contains two property values "FirstName", "LastName".
+
+## Expanded Entity Specification for OData V4.01
+In OData V4.01, a different [format](http://docs.oasis-open.org/odata/odata/v4.01/cs01/part1-protocol/odata-v4.01-cs01-part1-protocol.html#sec_ExpandedEntity) is introduced for $expand clause (for expanded entity and projected expanded entity).
```

Comment:
4.01 is not really a different format; more accurate would be to say that 4.01 differentiates between a selected navigation property (which appears without parens) and an expanded navigation property (which appears suffixed with parens, even if there are no nested expand options).